cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -398,12 +398,19 @@ namespace {
 Status CheckS3Initialized() {
   if (!IsS3Initialized()) {
     return Status::Invalid(
-        "S3 subsystem not initialized; please call InitializeS3() "
+        "S3 subsystem is not initialized; please call InitializeS3() "
         "before carrying out any S3-related operation");
   }
   return Status::OK();
 }
 
+Status CheckS3Finalized() {
```

Comment:
Is this actually useful? `is_initialized_` is set to false when finalizing, so `CheckS3Initialized` will already fail.

Reply:
In any case, please call this `CheckS3NotFinalized` as the current name is misleading.

Reply:
Ah, you're right.
We don't need this and I should have added `Not`...